### PR TITLE
Adequacy patch refactoring

### PR DESCRIPTION
**Pull request by payetvin**
My branch to refactor the adequacy patch
Branched from PR #961
Fixing the PR #903 comments

### CODE REVIEW

File: src/solver/optimisation/opt_fonctions.h [hunk 1]
```diff
@@ -30,6 +30,7 @@
 #include "../config.h"
 #include "opt_structure_probleme_a_resoudre.h"
 #include "../simulation/sim_structure_donnees.h"
+#include "../optimisation/adequacy_patch.h"
```
Comment (guilpier-code) [inline src/solver/optimisation/opt_fonctions.h]: If we don't add this include, it does not compile ?

Comment (payetvin) [reply src/solver/optimisation/opt_fonctions.h]: No the HOURLY_CSR_PROBLEM class is declared in this header

Comment (guilpier-code) [reply src/solver/optimisation/opt_fonctions.h]: ok, thanks

**Review by guilpier-code — APPROVED** [review]

File: src/libs/antares/study/parameters.cpp [hunk 2]
```diff
@@ -269,22 +269,32 @@ void Parameters::resetSeeds()
...
     adqPatch.curtailmentSharing.includeHurdleCost = false;
     adqPatch.curtailmentSharing.checkCsrCostFunction = false;
+}
+
+void Parameters::resetAdqPatchParameters()
+{
+    adqPatch.enabled = false;
+    resetAdqPatch_LocalMatching();
+    resetAdqPatch_CurtailmentSharing();
     resetThresholdsAdqPatch();
```
Comment (flomnes) [inline src/libs/antares/study/parameters.cpp:294]:
```cpp
adqPatch.localMatching.reset();
adqPatch.curtailmentSharing.reset(); // reset thresholds here
```

**Review by flomnes — COMMENTED** [review]
LGTM, just move all thresholds to curtailmentSharing.

**Review by flomnes — APPROVED** [review]

**Pull request merged**